### PR TITLE
feat: new lint-rule.md rule, 'header_comment"

### PR DESCRIPTION
**Pull request by a-frantz**
Pretty straight-forward rule. Essentially, `##` comments are neato and you should use them! **But only in the header** all other places these are found will get a warning.

### CODE REVIEW

File: lint-rules.md [hunk 1]
```diff
@@ -381,6 +381,14 @@ Bad:
     Int z = x + y# NO spaces between code and comment is hard to read.
 ```
 
+### `header_comment`
+
+Full line comments which start with a double-pound-sign (`##`) are given special treatment. They are called header-comments, and **should only be used at the very beginning of a file**. These special comments are meant as a place for documentation that doesn't belong in any of the language-supported meta sections. This is often whole file documentation or documentation for structs contained in the file.
+
+`sprocket doc` is under active development at the time of writing. This tool will auto-document any WDL file in MarkDown. Header-comments found by `sprocket doc` will be re-written in the output `.md` files (without the `##` prefix). Arbitrary and multi-line MarkDown constructions will be preserved adn rendered.
```
Comment (adthrasher) [inline lint-rules.md:388]:
```suggestion
`sprocket doc` is under active development at the time of writing. This tool will auto-document any WDL file in Markdown. Header-comments found by `sprocket doc` will be re-written in the output `.md` files (without the `##` prefix). Arbitrary and multi-line Markdown constructions will be preserved and rendered.
```

File: lint-rules.md [hunk 2]
```diff
@@ -381,6 +381,14 @@ Bad:
...
+
+These header-comments should not be used anywhere beside the header.
```
Comment (adthrasher) [inline lint-rules.md:390]:
```suggestion
These header-comments should not be used anywhere other than the header.
```

Comment (adthrasher) [reply lint-rules.md:390]: Maybe prefix with "As the name suggests, "

**Review by adthrasher — CHANGES_REQUESTED** [review]

**Comment by a-frantz** [comment]
@adthrasher IDK why the CI lint is failing, but IMO if all the code-tests are passing it's fine to merge this doc change. I think it would be inappropriate to make code changes in this PR.

File: lint-rules.md [hunk 3]
```diff
@@ -381,6 +381,14 @@ Bad:
     Int z = x + y# NO spaces between code and comment is hard to read.
 ```
 
+### `header_comment`
+
+Full line comments which start with a double-pound-sign (`##`) are given special treatment. They are called header-comments, and **should only be used at the very beginning of a file**. These special comments are meant as a place for documentation that doesn't belong in any of the language-supported meta sections. This is often whole file documentation or documentation for structs contained in the file.
+
+`sprocket doc` is under active development at the time of writing. This tool will auto-document any WDL file in Markdown. Header-comments found by `sprocket doc` will be re-written in the output `.md` files (without the `##` prefix). Arbitrary and multi-line Markdown constructions will be preserved and rendered.
+
+These header-comments should not be used anywhere other than the header.
```
[the remark this answers is not in the document]
Comment (a-frantz) [reply lint-rules.md:390]: Good point. I'll make that edit.

**Review by adthrasher — APPROVED** [review]

**Pull request merged**